### PR TITLE
Add tests for <Catalog /> and <CatalogEntityDetails />

### PR DESCRIPTION
**Pull request by chenhunghan**
- Set `safeDescriptors:false` in jest setup to be able to use `jest.spyOn`
- Add tests to `<Catalog />` and `<CatalogEntityDetails />`
- Bug fixes for `<Icon />` and `<CatalogMenu />` (found during adding tests)

### CODE REVIEW

File: src/renderer/components/icon/icon.tsx [hunk 1]
```diff
@@ -121,7 +121,7 @@ export class Icon extends React.PureComponent<IconProps> {
     }
 
     // render as material-icon
-    if (material) {
+    if (material && typeof material === "string") {
```
Comment (jweak) [inline src/renderer/components/icon/icon.tsx]: Why is this check needed? Isn't the type of `material` string or undefined so it should always be a string if it's defined?

Comment (chenhunghan) [reply src/renderer/components/icon/icon.tsx]: yeah, if users (extension developer) are using typescript (some just use javascript). In my test, I have tried to mock a category which has `category.metadata.icon = {}` and managed to pass `{}` to material props here (and crash the component) 

https://github.com/lensapp/lens/blob/da3c8d00c7ebe96c67b8c1be09878b4234ba6bf6/src/renderer/components/%2Bcatalog/catalog-menu.tsx#L48

File: src/renderer/components/+catalog/catalog-entity-details.test.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+import { mockWindow } from "../../../../__mocks__/windowMock";
+ 
+// avoid TypeError: window.requestIdleCallback is not a function
+mockWindow();
+ 
+describe("<CatalogEntityDetails />", () => {
+ 
+  beforeEach(() => {
+    CatalogEntityDetailRegistry.createInstance();
+  });
```
Comment (Nokel81) [inline src/renderer/components/+catalog/catalog-entity-details.test.tsx:36]: You should reset the instance after each test as well. Just in case we want to add more tests in the future.

File: src/renderer/components/+catalog/catalog-menu.tsx [hunk 3]
```diff
@@ -43,7 +43,7 @@ function getCategories() {
 function getCategoryIcon(category: CatalogCategory) {
   if (!category.metadata?.icon) return null;
 
-  return category.metadata.icon.includes("<svg")
+  return category.metadata.icon.includes?.("<svg")
```
Comment (Nokel81) [inline src/renderer/components/+catalog/catalog-menu.tsx:46]: why is this here? would it be better to just check if the `typeof` above is correct?

File: src/renderer/components/icon/icon.tsx (hunk 1 above)
Comment (Nokel81) [reply src/renderer/components/icon/icon.tsx]: shouldn't you also do the check for `svg`? And wouldn't it be good enough to just do a `typeof` check?

Comment (chenhunghan) [reply src/renderer/components/icon/icon.tsx]: That's true, thanks.

File: src/renderer/components/+catalog/catalog-entity-details.test.tsx (hunk 2 above)
Comment (chenhunghan) [reply src/renderer/components/+catalog/catalog-entity-details.test.tsx:36]: Added https://github.com/lensapp/lens/pull/3927/commits/a89ef63477b2fabfdda0dd12689e6e2fbfcf7f65

File: src/renderer/components/icon/icon.tsx (hunk 1 above)
Comment (chenhunghan) [reply src/renderer/components/icon/icon.tsx]: Added https://github.com/lensapp/lens/pull/3927/commits/c55b2fa1ae0cec757985c30b2bf4f8058b1ed9f3

File: src/renderer/components/+catalog/catalog-menu.tsx (hunk 3 above)
Comment (chenhunghan) [reply src/renderer/components/+catalog/catalog-menu.tsx:46]: Not sure that (adding `typeof` checks to `category.metadata?.icon`), will be the original intent of the code, there was no spec. But I guess both should be fine...

**Comment by Nokel81** [comment]
Why did you close this?
